pkg/handlers/generic/mutation/controlplanevirtualip/inject.go

```diff
@@ -117,23 +95,27 @@ func (h *ControlPlaneVirtualIP) Mutate(
 		return err
 	}
 
-	var virtualIPProvider providers.Provider
-	// only kube-vip is supported, but more providers can be added in the future
-	if controlPlaneEndpointVar.VirtualIPSpec.Provider == v1alpha1.VirtualIPProviderKubeVIP {
-		virtualIPProvider = providers.NewKubeVIPFromConfigMapProvider(
-			h.client,
-			h.config.defaultKubeVIPConfigMapName,
-			h.config.DefaultsNamespace(),
-		)
-	}
-
 	return patches.MutateIfApplicable(
 		obj,
 		vars,
 		&holderRef,
 		selectors.ControlPlane(),
 		log,
 		func(obj *controlplanev1.KubeadmControlPlaneTemplate) error {
+			if controlPlaneEndpointVar.VirtualIPSpec == nil {
+				log.V(5).Info("ControlPlane VirtualIP not set")
+				// if VirtualIPSpec is not set, delete all template files
+				// as we do not want them to end up in the generated KCP
+				deleteFiles(obj, providers.TemplateFileNames...)
```

Comment:
This is actually removing only KubeVIP related files only right?
the comment should reflect that.
Also it would be nice to rename `TemplateFileNames` to `VirtualIPProviderFileNames`

Reply:
Makes sense, I didn't include "KubeVIP" here because by this points in the code its dealing with the virtualIPProvider interface, but updated the comment with more details, PTAL.

Reply:
Renamed the varibale too, thanks for the suggestion.